launcher/modplatform/import_ftb/PackHelpers.cpp

```diff
@@ -0,0 +1,87 @@
+// SPDX-License-Identifier: GPL-3.0-only
+/*
+ *  Prism Launcher - Minecraft Launcher
+ *  Copyright (c) 2023 Trial97 <alexandru.tripon97@gmail.com>
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, version 3.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+#include "modplatform/import_ftb/PackHelpers.h"
+
+#include <QIcon>
+#include <QString>
+#include <QVariant>
+
+#include "FileSystem.h"
+#include "Json.h"
+
+namespace FTBImportAPP {
+
+Modpack parseDirectory(QString path)
+{
+    Modpack modpack{ path };
+    auto instanceFile = QFileInfo(FS::PathCombine(path, "instance.json"));
+    if (!instanceFile.exists() || !instanceFile.isFile())
+        return {};
+    try {
+        auto doc = Json::requireDocument(instanceFile.absoluteFilePath(), "FTB_APP instance JSON file");
+        const auto root = doc.object();
+        modpack.uuid = Json::requireString(root, "uuid", "uuid");
+        modpack.id = Json::requireInteger(root, "id", "id");
+        modpack.versionId = Json::requireInteger(root, "versionId", "versionId");
+        modpack.name = Json::requireString(root, "name", "name");
+        modpack.version = Json::requireString(root, "version", "version");
+        modpack.mcVersion = Json::requireString(root, "mcVersion", "mcVersion");
+        modpack.jvmArgs = Json::requireVariant(root, "jvmArgs", "jvmArgs");
```

Comment:
This will throw if `jvmArgs` is null. i'm guessing you don't want that!

Reply:
Thanks for the catch.
Solved but right now I do not see my commits on the PR(they are on the branch) so I think maybe GitHub has a hiccup.